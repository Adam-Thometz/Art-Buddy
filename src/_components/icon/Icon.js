import React from "react";

import './Icon.css';

const Icon = ({ icon, text, onClick = null, size = "250px", width = null, largeFont = false,  id = null }) => {
  const imgStyles = {
    height: size,
    width: width ? width : size,
    cursor: onClick ? 'pointer' : null
  };

  const textStyles = {
    fontSize: largeFont ? '2.5rem' : (parseInt(size) <= 50 ? '1rem' : '1.75rem')
  };

  return (
    <figure className="Icon" onClick={onClick} id={id ? id : null}>
      <img style={imgStyles} src={icon} alt="" />
      <figcaption className="Icon-text" style={textStyles}>{text}</figcaption>
    </figure>
  );
};

export default Icon;