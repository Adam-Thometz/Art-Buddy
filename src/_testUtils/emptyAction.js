const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION