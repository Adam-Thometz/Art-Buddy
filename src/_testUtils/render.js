import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
