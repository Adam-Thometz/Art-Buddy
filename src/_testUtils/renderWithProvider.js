import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;