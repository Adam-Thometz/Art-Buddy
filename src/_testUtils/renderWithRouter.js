import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;