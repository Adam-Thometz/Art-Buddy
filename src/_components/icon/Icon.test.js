import Icon from './Icon';

import { render } from '_testUtils/render';

describe('Icon component', () => {
  it('renders without crashing', () => {
    render(<Icon />);
  });

  it('matches the snapshot of the icon', () => {
    const { asFragment } = render(<Icon />);
    expect(asFragment()).toMatchSnapshot();
  });
});