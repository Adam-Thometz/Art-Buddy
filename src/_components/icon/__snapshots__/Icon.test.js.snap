// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clickable component matches the snapshot of the clickable icon 1`] = `
<DocumentFragment>
  <div
    class="Icon"
  >
    <img
      alt=""
      style="width: 250px; height: 250px;"
    />
    <p />
  </div>
</DocumentFragment>
`;
